### ISSUE TITLE
Bu proje, deniz suyundan hidrojen yakıtı üretimi ve uzay itki sistemlerinde kullanımı üzerine odaklanmıştır. Proje kapsamında, bor katkısının enerji verimliliği üzerindeki etkisi de incelenmektedir. Aynı zamanda, deniz suyunun radyasyon kalkanı olarak kullanımı araştırılmaktadır

### ISSUE THREAD

**Issue by emrahuzucar355** [emrahuzucar355/Uzucar-#5]
_Originally posted by @emrahuzucar355 in https://github.com/emrahuzucar355/Uzucar-/issues/4_

import numpy as np
import matplotlib.pyplot as plt

# Sabit Değerler
gamma = 1.4  # Gaz sabiti oranı
R = 287  # Gaz sabiti (J/kg.K)
Tc = 3500  # Yanma odası sıcaklığı (K)
mdot = 10  # Kütle akışı (kg/s)
Pc = 2e6  # Yanma odası basıncı (Pa)
Ae = 0.1  # Nozul çıkış alanı (m^2)
Pe = 10000  # Nozul çıkış basıncı (Pa)
bor_katkisi_baslangic = 10  # Bor katkısı başlangıç değeri (%)

# Fonksiyonlar
def enerji_verimliligi(bor_katkisi):
    """
    Bor katkısına bağlı enerji verimliliği hesaplama.
    Başlangıç verimliliği %85, bor katkısı başına %0.375 artış.
    """
    return 85 + bor_katkisi * 0.375

def enerji_tasarrufu(verimlilik):
    """
    Verimliliğe bağlı enerji tasarrufu hesaplama.
    Tasarruf başlangıç verimliliği (%85) ile karşılaştırılır.
    """
    return (verimlilik - 85) / 7.5

# Bor katkısına göre değerlerin hesaplanması
bor_degerleri = np.linspace(0, 20, 100)  # Bor katkısı %0 ile %20 arasında
verimlilik_degerleri = enerji_verimliligi(bor_degerleri)
tasarruf_degerleri = enerji_tasarrufu(verimlilik_degerleri)

# Belirli bir bor katkısı için hesaplamalar (%10)
bor_katkisi_orani = bor_katkisi_baslangic
toplam_verimlilik = enerji_verimliligi(bor_katkisi_orani)
toplam_tasarruf = enerji_tasarrufu(toplam_verimlilik)

# Sonuçların Yazdırılması
print(f"Bor Katkısı: {bor_katkisi_orani:.2f}%")
print(f"Toplam Verimlilik: {toplam_verimlilik:.2f}%")
print(f"Toplam Enerji Tasarrufu: {toplam_tasarruf:.2f}%")

# Grafik Çizimi
plt.figure(figsize=(12, 6))
plt.plot(bor_degerleri, verimlilik_degerleri, label="Verimlilik (%)", color="blue", linewidth=2)
plt.plot(bor_degerleri, tasarruf_degerleri, label="Enerji Tasarrufu (%)", color="green", linewidth=2)
plt.scatter(bor_katkisi_orani, toplam_verimlilik, color="red", label=f"Bor Katkısı: %{bor_katkisi_orani}", zorder=5)
plt.axhline(85, color="gray", linestyle="--", label="Başlangıç Verimliliği (%85)")
plt.title("Bor Katkısının Enerji Verimliliği ve Tasarruf Üzerine Etkisi", fontsize=14, fontweight="bold")
plt.xlabel("Bor Katkısı (%)", fontsize=12)
plt.ylabel("Yüzde (%)", fontsize=12)
plt.grid(alpha=0.3)
plt.legend(fontsize=10)
plt.tight_layout()
plt.show()

**Comment by emrahuzucar355** [comment]
import numpy as np
import matplotlib.pyplot as plt

# Sabitler
g = 9.81  # Yerçekimi ivmesi (m/s²)
isp = 450  # Spesifik impuls (saniye)
fuel_mass = 100000  # Başlangıç yakıt kütlesi (kg)
dry_mass = 20000  # Gemi kuru kütlesi (kg)
engine_power = 5000  # Motor gücü (kW)
radiation_limit = 2.0  # Radyasyon limiti (Sv/saat)
energy_efficiency = 0.8  # Enerji sistem verimliliği

# Simülasyon parametreleri
time = np.linspace(0, 5000, 1000)  # Zaman (saniye)
dt = time[1] - time[0]  # Zaman adımı
fuel = np.zeros_like(time)  # Yakıt miktarı
speed = np.zeros_like(time)  # Hız
radiation = np.zeros_like(time)  # Radyasyon seviyesi
energy = np.zeros_like(time)  # Enerji seviyesi
oxygen = np.zeros_like(time)  # Oksijen seviyesi
fuel[0] = fuel_mass
energy[0] = engine_power * energy_efficiency  # Başlangıç enerjisi
oxygen[0] = 100  # Başlangıç oksijen seviyesi (örnek birim)

# Simülasyon
for i in range(1, len(time)):
    if fuel[i - 1] > 0:  # Yakıt varsa
        fuel_consumption = engine_power * dt / (isp * g)  # Yakıt tüketimi (kg)
        fuel[i] = fuel[i - 1] - fuel_consumption
        if fuel[i] < 0:  # Yakıt tükenirse sıfırla
            fuel[i] = 0
        thrust = isp * g * fuel_consumption  # İtki gücü (N)
        acceleration = thrust / (dry_mass + fuel[i])  # İvme (m/s²)
        speed[i] = speed[i - 1] + acceleration * dt  # Hız güncellemesi
    else:
        speed[i] = speed[i - 1]  # Yakıt yoksa hız sabit

    # Enerji seviyesi
    energy_loss = engine_power * (1 - energy_efficiency) * dt  # Enerji kaybı
    energy[i] = energy[i - 1] - energy_loss
    if energy[i] < 0:  # Enerji tükenirse sıfırla
        energy[i] = 0

    # Radyasyon seviyesi
    radiation[i] = radiation[i - 1] + 0.01 * speed[i] * dt  # Hızla artan radyasyon

    # Oksijen tüketimi
    oxygen_consumption = 0.001 * dt  # Oksijen tüketim oranı
    oxygen[i] = oxygen[i - 1] - oxygen_consumption
    if oxygen[i] < 0:  # Oksijen tükenirse sıfırla
        oxygen[i] = 0

# Grafikler
plt.figure(figsize=(12, 8))

# Yakıt, Hız ve Enerji
plt.subplot(3, 1, 1)
plt.plot(time, fuel, label="Yakıt (kg)", color='blue')
plt.plot(time, energy, label="Enerji (kW)", color='green')
plt.title("Yakıt ve Enerji Yönetimi")
plt.xlabel("Zaman (saniye)")
plt.ylabel("Miktar")
plt.legend()

# Hız
plt.subplot(3, 1, 2)
plt.plot(time, speed, label="Hız (m/s)", color='orange')
plt.title("Gemi Hızlanması")
plt.xlabel("Zaman (saniye)")
plt.ylabel("Hız (m/s)")
plt.legend()

# Radyasyon ve Oksijen
plt.subplot(3, 1, 3)
plt.plot(time, radiation, label="Radyasyon Seviyesi (Sv)", color='red')
plt.plot(time, oxygen, label="Oksijen Seviyesi", color='purple')
plt.title("Radyasyon ve Oksijen Yönetimi")
plt.xlabel("Zaman (saniye)")
plt.ylabel("Miktar")
plt.legend()

plt.tight_layout()
plt.show()
